frontend/views/containers/payments/PaymentDetail.vue

```diff
@@ -114,9 +112,11 @@ export default ({
         sbp('okTurtles.events/emit', SET_MODAL_QUERIES, 'PaymentDetail', { id })
       }
       if (payment) {
-        this.payment = cloneDeep(payment)
         // TODO: the payment augmentation duplication in Payment and PaymentRecord, and between todo/sent/received, needs to be resolved more thoroughly
-        this.payment.periodstamp = this.periodStampGivenDate(this.payment.meta.createdDate)
+        this.payment = {
+          ...cloneDeep(payment),
+          periodstamp: await this.historicalPeriodStampGivenDate(payment.meta.createdDate)
+        }
```

Comment:
If you're going to combine lines 117 and 119 like this, you might as well combine line 120 into here too to be consistent.

Reply:
OK - by the way, not sure whether `comparePeriodStamps` is necessary, since period stamps are not abstract IDs but always timestamps, which support relational operators out-of-the-box (unlike abstract IDs which may or may not be ordered)